src/pages/iou/request/step/IOURequestStepCategory.js

```diff
@@ -49,11 +62,22 @@ function IOURequestStepCategory({
      * @param {String} category.searchText
      */
     const updateCategory = (category) => {
-        if (category.searchText === transaction.category) {
-            IOU.resetMoneyRequestCategory_temporaryForRefactor(transactionID);
-        } else {
-            IOU.setMoneyRequestCategory_temporaryForRefactor(transactionID, category.searchText);
+        const isSelectedCategory = category.searchText === transactionCategory;
+        const updatedCategory = isSelectedCategory ? '' : category.searchText;
+
+        // The case edit split bill
+        if (isSplitBill && isEditing) {
+            IOU.setDraftSplitTransaction(transaction.transactionID, {category: category.searchText});
```

Comment:
Do we allow disable category in this case? Why don't you use updatedCategory here?